cmake/external/pybind11.cmake

```diff
@@ -1,24 +1,13 @@
-
-if(onnxruntime_PREFER_SYSTEM_LIB)
-  find_package(pybind11)
-endif()
-
-if(NOT TARGET pybind11::module)
-  include(ExternalProject)
-
-  set(pybind11_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/pybind11/src/pybind11/include)
-  set(pybind11_URL https://github.com/pybind/pybind11.git)
-  set(pybind11_TAG v2.6.2)
-
-  ExternalProject_Add(pybind11
-        PREFIX pybind11
-        GIT_REPOSITORY ${pybind11_URL}
-        GIT_TAG ${pybind11_TAG}
-        CONFIGURE_COMMAND ""
-        BUILD_COMMAND ""
-        INSTALL_COMMAND ""
-  )
-  set(pybind11_dep pybind11)
-else()
+FetchContent_Declare(
+    pybind11_project
+    URL ${DEP_URL_pybind11}
+	URL_HASH SHA1=${DEP_SHA1_pybind11}
+	FIND_PACKAGE_ARGS 2.6 NAMES pybind11
+)
+FetchContent_MakeAvailable(pybind11_project)
+
+if(TARGET pybind11::module)
   set(pybind11_lib pybind11::module)
+else()
+  set(pybind11_dep pybind11::pybind11)
```

Comment:
>set(pybind11_dep pybind11::pybind11) [](http://example.com/codeflow?start=2&length=36)

Can we do something to make other 3rd parties to use our pybind?

Reply:
Yes. For example, they may also have FetchContent_Declare/FetchContent_MakeAvailable.  See https://cmake.org/cmake/help/latest/module/FetchContent.html#id9 for details.